src/bootstrap/src/core/builder.rs

```diff
@@ -1667,21 +1669,20 @@ impl<'a> Builder<'a> {
             }
         }
 
-        if let Some(host_linker) = self.linker(compiler.host) {
-            hostflags.arg(format!("-Clinker={}", host_linker.display()));
-        }
-        if self.is_fuse_ld_lld(compiler.host) {
-            hostflags.arg("-Clink-args=-fuse-ld=lld");
-        }
+        linker_args(self, compiler.host, LldThreads::Yes).into_iter().for_each(|flag| {
+            hostflags.arg(flag);
+        });
 
         if let Some(target_linker) = self.linker(target) {
             let target = crate::envify(&target.triple);
             cargo.env(&format!("CARGO_TARGET_{target}_LINKER"), target_linker);
         }
-        if self.is_fuse_ld_lld(target) {
-            rustflags.arg("-Clink-args=-fuse-ld=lld");
-        }
-        self.lld_flags(target).for_each(|flag| {
+        // We want to set -Clinker using Cargo, therefore we only call `linker_flags` and not
+        // `linker_args` here.
+        linker_flags(self, target, LldThreads::Yes).into_iter().for_each(|flag| {
+            rustflags.arg(&flag);
+        });
+        linker_args(self, target, LldThreads::Yes).into_iter().for_each(|flag| {
```

Comment:
`lld_flags` previously behave like `linker_flags`.
Is it intentional change to add `-Clinker` here?

Reply:
In this case, I think we should add it? Since it's not being added elsewhere, unless `CARGO_TARGET_<target>_LINKER` applies to rustdoc.

Reply:
```
$ CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=lldx cargo test --doc
```
doesn't seem to fail, so this environment variabe probably isn't applied to `rustdoc`. Therefore, I think that we should keep `args` and not `flags`.